src/include/stir/recon_buildblock/BackProjectorByBin.h

```diff
@@ -111,14 +112,42 @@ class BackProjectorByBin :
 		   const RelatedViewgrams<float>&,
 		   const int min_axial_pos_num, const int max_axial_pos_num,
 		   const int min_tangential_pos_num, const int max_tangential_pos_num);
+#endif
+ //! project whole proj_data into the volume
+ /*! it overwrites the data already present in the volume */
+ virtual void back_project(const ProjData&, int subset_num = 0, int num_subsets = 1);
+
+ /*! \brief projects the viewgrams into the volume
+  it adds to the data already present in the volume.*/
+void back_project(const RelatedViewgrams<float>&);
+
+ /*! \brief projects the specified range of the viewgrams into the volume
+  it adds to the data already present in the volume.*/
+void back_project(const RelatedViewgrams<float>&,
+          const int min_axial_pos_num, const int max_axial_pos_num);
+
+ /*! \brief projects the specified range of the viewgrams into the volume
+   it adds to the data already present in the volume.*/
+void back_project(const RelatedViewgrams<float>&,
+          const int min_axial_pos_num, const int max_axial_pos_num,
+          const int min_tangential_pos_num, const int max_tangential_pos_num);
 
+ /*! \brief tell the back projector to start accumulating into a new image*/
+ virtual void start_accumulating_in_new_target();
+
+ /// Get output
+ virtual void get_output(DiscretisedDensity<3,float> &) const;
 
 protected:
 
   virtual void actual_back_project(DiscretisedDensity<3,float>&,
                                    const RelatedViewgrams<float>&,
 		                   const int min_axial_pos_num, const int max_axial_pos_num,
-		                   const int min_tangential_pos_num, const int max_tangential_pos_num) = 0;
+                           const int min_tangential_pos_num, const int max_tangential_pos_num);
+
+ virtual void actual_back_project(const RelatedViewgrams<float>&,
```

Comment:
we need some documentation on what these 2 do and what should be overriden

Reply:
How about this:

```
  /*! \brief This actually does the back projection.
   There are two versions of this code to enable backwards compatibility.

   This is the newer version (in which the backprojected image is a member variable).
   In most cases, the new version calls the old version (in which the backprojected image is not a member variable).

   If you are developing your own projector, one of these two needs to be overloaded. It doesn't matter which,
   but it might as well be the new one in case we one day decide to remove the old ones.
  */
```